modules/vsphere-load-balancer-multiple-subnets.adoc

```diff
@@ -0,0 +1,112 @@
+====
+
+.Prerequisites
+* On your load balancer, TCP over ports 22623, 6443, 443, and 80 must be reachable by all nodes in your cluster.
+* Load balance the API port, 6443, between each of the control plane nodes.
+
+* Load balance the application ports, 443 and 80, between all the compute nodes.
+
+* On your load balancer, port 22623, which is used to serve ignition startup configurations to nodes, is not exposed outside the cluster.
+
+* Your load balancer can access the required ports on each node in your cluster. You can ensure this level of access by completing the following actions: 
+** The API load balancer can access ports 22623 and 6443 on the control plane nodes.
+** The ingress load balancer can access ports 443 and 80 on the nodes where the ingress pods are located.
+
+.Procedure
+
+. Configure an `/etc/haproxy/haproxy.cfg` configuration for an HAProxy load balancer, so that the local balancer can access multiple subnets in your VMware vSphere environment.
++
+The following example API and application ingress load balancer demonstrates a configured backend `api-server`
+
+.Sample API and application ingress load balancer configuration
+[%collapsible]
+[source,text]
+----
+global
+  log         127.0.0.1 local2
+  pidfile     /var/run/haproxy.pid
+  maxconn     4000
+  daemon
+defaults
+  mode                    http
+  log                     global
+  option                  dontlognull
+  option http-server-close
+  option                  redispatch
+  retries                 3
+  timeout http-request    10s
+  timeout queue           1m
+  timeout connect         10s
+  timeout client          1m
+  timeout server          1m
+  timeout http-keep-alive 10s
+  timeout check           10s
+  maxconn                 3000
+frontend stats
+  bind *:1936
+  mode            http
+  log             global
+  maxconn 10
+  stats enable
+  stats hide-version
+  stats refresh 30s
+  stats show-node
+  stats show-desc Stats for ocp4 cluster
+  stats auth admin:ocp4
+  stats uri /stats
+backend api-server <1>
+  option  httpchk GET /readyz HTTP/1.0
+  option  log-health-checks
+  balance roundrobin
+  %{ for addr in api ~}
+  server ${addr} ${addr}:6443 weight 1 verify none check check-ssl inter 1s fall 2 rise 3
+  %{ endfor ~}
+listen api-server-6443
+  bind *:6443
+  mode tcp
+  server bootstrap bootstrap.ocp4.example.com:6443 check inter 1s backup
```

Comment:
I am thinking we don't provide an example. This is going to be very difficult to convey and I am sure a customer most likely won't be using haproxy.

Each listen section needs the entire IP range (or at least the DHCP range) for each subnet in which a node type will reside.

Example from CI:
https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/conf/vsphere/lb/external/ipi-conf-vsphere-lb-external-commands.sh#L70-L116

@rvanderp3 wdyt?

Reply:
I'm happy with this approach, because a procedure does not seem viable for all the possible use cases.